client/app/reducers/nav/index.js

```diff
@@ -0,0 +1,16 @@
+import { NAV } from '../../constants/nav'
+import Immutable from 'seamless-immutable'
+const initialState = Immutable({
+  services: 0,
+  orders: 0,
+  catalogs: 0
+})
+
+export function NavReducer (state = initialState, action) {
+  switch (action.type) {
+    case NAV.ADD_COUNT:
+      return state.set(action.itemType, action.count)
```

Comment:
Instead of having a generic `ADD_COUNT` action type, consider being more specific, it can have positive impact on code readability, e.g. `ADD_SERVICE_COUNT`.

Alternatively, consider being more specific in terms of your action creator functions, for example:

```
export const addServiceCount = (count) => ({
  type: NAV.ADD_COUNT,
  itemType: 'services',
  count: count
})
```

This way, your code can do `addServiceCount(1)` instead of `addCount('services', 1)` or `addCount(constants.services, 1)`.

More generally, your Redux action creators and action types are the contract between your code and the Redux infra. Therefore, these things should be as sensible as possible, which in turn makes your code itself more sensible.

Reply:
Thank you for the feedback on this.  I think that in this particular case I weighed your suggestion and also weighed the notion that creating a list of those other actions would make the code more verbose but the actions each has would be identical.  I think in this case just for this instance it makes sense to keep the code down to a minimum rather than creating 4+ exactly identical actions.